src/caches/beliefpropagationcache.jl

```diff
@@ -208,7 +226,7 @@ function update(
   verbose=false,
   kwargs...,
 )
-  compute_error = !isnothing(tol)
+  compute_error = !isnothing(tol) && !hasqns(tensornetwork(bp_cache))
```

Comment:
Is this just temporarily circumvent a QN issue?

Reply:
Yeah, the way we define `message_diff()` doesn't work for `QN` tensors at the moment, so I added that to avoid it.

Reply:
I see, I think it would be better to address the issue in `message_diff`.

Reply:
Okay I have addressed it below and will resolve this one.